substrate/frame/revive/src/vm/evm/instructions/macros.rs

```diff
@@ -122,51 +122,19 @@ macro_rules! gas_or_fail_legacy {
 	};
 }
 
-use crate::vm::Ext;
-use revm::interpreter::gas::{MemoryExtensionResult, MemoryGas};
-
-/// Adapted from
-/// https://docs.rs/revm/latest/revm/interpreter/struct.Gas.html#method.record_memory_expansion
-pub fn record_memory_expansion<E: Ext>(
-	memory: &mut MemoryGas,
-	_ext: &mut E,
-	new_len: usize,
-) -> MemoryExtensionResult {
-	let Some(_additional_cost) = memory.record_new_len(new_len) else {
-		return MemoryExtensionResult::Same;
-	};
-
-	// TODO: Commented this for now so that extending memory will not cost any gas.
-	// if ext.gas_meter_mut().charge_evm_gas(additional_cost).is_err() {
-	// 	return MemoryExtensionResult::OutOfGas;
-	// }
-
-	MemoryExtensionResult::Extended
-}
-
-/// Resizes the interpreterreter memory if necessary. Fails the instruction if the memory or gas
-/// limit is exceeded.
+/// Ensures that memory access at `offset..offset+len` is valid.
+/// Revert if the memory is out of bounds.
 #[macro_export]
-macro_rules! resize_memory {
+macro_rules! check_memory_bounds {
 	($interpreter:expr, $offset:expr, $len:expr) => {
-		resize_memory!($interpreter, $offset, $len, ())
+		check_memory_bounds!($interpreter, $offset, $len, ())
 	};
 	($interpreter:expr, $offset:expr, $len:expr, $ret:expr) => {
-		let words_num = revm::interpreter::num_words($offset.saturating_add($len));
-		match crate::vm::evm::instructions::macros::record_memory_expansion(
-			$interpreter.gas.memory_mut(),
-			$interpreter.extend,
-			words_num,
-		) {
-			revm::interpreter::gas::MemoryExtensionResult::Extended => {
-				$interpreter.memory.resize(words_num * 32);
-			},
-			revm::interpreter::gas::MemoryExtensionResult::OutOfGas => {
-				$interpreter.halt(revm::interpreter::InstructionResult::MemoryOOG);
-				return $ret;
-			},
-			revm::interpreter::gas::MemoryExtensionResult::Same => (), // no action
-		};
+		if $offset.saturating_add($len) > $interpreter.memory.len() {
+			log::debug!(target: $crate::LOG_TARGET, "check memory bounds failed: offset={} len={} memory_size={}", $offset, $len, $interpreter.memory.len());
+			$interpreter.halt(revm::interpreter::InstructionResult::Revert);
```

Comment:
```suggestion
			$interpreter.halt(revm::interpreter::InstructionResult::MemoryOOG);
```

Reply:
yeah was not sure about that, it's not really an out of gas error, it's more a OOM

Reply:
But we should return the same error that EVM returns when you try to access out of bound memory. Even though its not due to to gas but a fixed limit. I agree with cyrill here.